_posts/2022-04-01-introducing-pair-programming.adoc

```diff
@@ -0,0 +1,113 @@
+---
+title: Pair programming using VS Code extension
+layout: post
+author: Florent Benoit
+description: >-
+  How to do pair programming using a VS Code extension.
+categories: []
+keywords: ['VSCode', 'codetogether', 'pair programming']
+slug: /@florent.benoit/pair-programming-using-vscode-editor
+---
+
+One of the most wanted feature on Eclipse Che is to be able to do pair programming.
+
+This blog post describes how it is possible to achieve that by using a third-party extension, link:https://open-vsx.org/extension/genuitecllc/codetogether[CodeTogether].
+
+There are different pricing plans available for this extension, but there is one free item when connecting to a maximum of 60 minutes and 3 shares which may be enough for most of the case.
+
+This is only possible using Eclipse Che starting with 7.46+ and using Visual Studio Code editor, not Che-Theia.
+
+
+== Quick start from an example
+
+. Go the dashboard page of Eclipse Che. For example URL will be following this format https://eclipse-che-my-instance.com/dashboard/#/
+
+. Replace the last part of the URL containing `/dashboard/#` by `#https://github.com/benoitf/codetogether-che-petclinic`. URL should look like `https://eclipse-che-my-instance.com#https://github.com/benoitf/codetogether-che-petclinic``
+
+. Press enter key to open this URL in the browser.
+
+. Workspace will start and it will open automatically Visual Studio Code using a sample project based on `Petclinic` application.
+
+. As link:https://open-vsx.org/extension/genuitecllc/codetogether[CodeTogether] is a recommended extension, accept the prompt asking to install this extension.
+
+. CodeTogether will ask to access Visual Studio Code, click on `No`
+
+. A new prompt will ask to redirect port 1593: Click on `No`
+
+. Bring the command palette using for example `F1` key and select `CodeTogether: Start Hosting Session`
+
+. Keep default on all properties and click on the `start` button
+
+. Click "No" on "CodeTogether requires access to Visual Studio Code Proposed APIs"
+
+. `Pair programming` link is then available in the clipboard.
+
+. Now, share this link with your team members. They will be able to join your workspace.
+
+. It is possible to join from different IDEs or from a running Eclipse Che workspace (like the one we are using). But click on joining a session rather than starting a new session: 'CodeTogether: Join Session' and paste the link in the input field.
```

Comment:
```suggestion
. It is possible to use this link to join from different IDEs or from a running Eclipse Che workspace (like the one we are using). To do that select 'CodeTogether: Join Session' rather than starting a new session.
```